libs/platform/src/lib/components/form/date-picker/date-picker.component.ts

```diff
@@ -0,0 +1,326 @@
+     * Special day number, list 1-20 [class:`fd-calendar__special-day--{{number}}`] is available there:
+     * https://sap.github.io/fundamental-styles/components/calendar.html calendar special days section
+     * Rule accepts method with FdDate object as a parameter. ex:
+     * `rule: (fdDate: FdDate) => fdDate.getDay() === 1`, which will mark all sundays as special day.
+     */
+    @Input()
+    specialDaysRules: SpecialDayRule<D>[] = [];
+
+    /**
+     * Object to customize year grid,
+     * Row, Columns and method to display year can be modified
+     */
+    @Input()
+    yearGrid: CalendarYearGrid = {
+        rows: 4,
+        cols: 5,
+        yearMapping: (num: number) => num.toString()
+    };
+
+    /**
+     * Object to customize aggregated year grid,
+     * Row, Columns and method to display year can be modified
+     */
+    @Input()
+    aggregatedYearGrid: CalendarYearGrid = {
+        rows: 4,
+        cols: 3,
+        yearMapping: (num: number) => num.toString()
+    };
+
+    /**
+     * Whether user wants to mark sunday/saturday with `fd-calendar__item--weekend` class
+     */
+    @Input()
+    markWeekends = true;
+
+    /**
+     * Whether user wants to show week numbers next to days
+     */
+    @Input()
+    showWeekNumbers = true;
+
+    /** Whether the date picker is open. Can be used through two-way binding. */
+    @Input()
+    isOpen = false;
+
+    /** Event emitted when the state of the isOpen property changes. */
+    @Output()
+    readonly isOpenChange = new EventEmitter<boolean>();
+
+    /** Fired when a new date is selected. */
+    @Output()
+    readonly selectedDateChange: EventEmitter<D> = new EventEmitter<D>();
+
+    /** Event thrown every time selected first or last date in range mode is changed */
+    @Output()
+    readonly selectedRangeDateChange: EventEmitter<DateRange<D>> = new EventEmitter<DateRange<D>>();
+
+    /** Event thrown every time calendar active view is changed */
+    @Output()
+    readonly activeViewChange: EventEmitter<FdCalendarView> = new EventEmitter<FdCalendarView>();
+
+    /**
+     * @hidden core date-picker as child
+     */
+    @ViewChild(FdDatePickerComponent)
+    fdDatePickerComponent: FdDatePickerComponent<D>;
+
+    private _state: Status;
+
+    /**
+     * Function used to disable certain dates in the calendar.
+     * @param _ D
+     */
+    @Input()
+    disableFunction = function (_: D): boolean {
+        return false;
+    };
+
+    /**
+     * Function used to disable certain dates in the calendar for the range start selection.
+     * @param _ D
+     */
+    @Input()
+    disableRangeStartFunction = function (_: D): boolean {
+        return false;
+    };
+
+    /**
+     * Function used to disable certain dates in the calendar for the range end selection.
+     * @param _ D
+     */
+    @Input()
+    disableRangeEndFunction = function (_: D): boolean {
+        return false;
+    };
```

Comment:
Still not by [Code Contribution Guideline](https://github.com/SAP/fundamental-ngx/wiki/Code-Contribution-Guideline) 🙃
Properties decorated by `@Input` should be above the not decorated, properties, decorated by `@Output` and other decorators

Reply:
These are methods. Getting linting error on moving it before @Output properties.